### PR TITLE
implement a skip_config_creation option in get_user_info

### LINKED ISSUE #244: implement a `skip_config_creation` option in `get_user_info`
### Problem
A user may want to user `get_user_info` but not want it to interrupt execution if it can't find a config file.b

### Proposed solution
as a NON DEFAULT option allow them to override interrupt.

This occured to @sbillinge while he was tweaking the docs.  He wrote the docs as if it were implemented.   Here is what he wrote:
```
Getting user data with no config files and with no interruption of execution
----------------------------------------------------------------------------

If you would like get run ``get_user_data()`` but without execution interruption even if it cannot find
an input file, type

.. code-block:: python

    user_data = get_user_data(skip_config_creation=True)
```
But this needs to be implemented

### PR DESCRIPTION
**Pull request by yucongalicechen**
closes #244 
@sbillinge ready for some feedback
btw I see in the docs you used `get_user_data` instead of `get_user_info` at a couple places, which seems to be a better name, shall we change to `get_user_data` instead?

### CODE REVIEW

File: tests/test_tools.py [hunk 1]
```diff
@@ -22,6 +22,9 @@ def _run_tests(inputs, expected):
     config = get_user_info(args)
     assert config.get("username") == expected_username
     assert config.get("email") == expected_email
+    config = get_user_info(args, skip_config_creation=True)
+    assert config.get("username") == expected_username
+    assert config.get("email") == expected_email
```
Comment (yucongalicechen) [inline tests/test_tools.py:27]: It looks like we can reuse this run_test function for most test cases for different `skip_config_creation` except when there're no inputs or files, so I added this here

**Comment by sbillinge** [comment]
> closes #244 @sbillinge ready for some feedback btw I see in the docs you used `get_user_data` instead of `get_user_info` at a couple places, which seems to be a better name, shall we change to `get_user_data` instead?

haha, that was a mistake on my part.    I am not sure tbh.  I am ok with either "user-data* or "user-info".   @bobleesj @yucongalicechen which do you think is more obvious?

File: tests/test_tools.py [hunk 2]
```diff
@@ -118,7 +121,20 @@ def test_get_user_info_no_conf_file_no_inputs(monkeypatch, inputsa, inputsb, exp
...
+    args = {"username": inputsa[0], "email": inputsa[1]}
+    expected_username, expected_email = expected
+
+    # Test with user inputs
+    config = get_user_info(args)
+    assert config.get("username") == expected_username
+    assert config.get("email") == expected_email
+
+    # Test skipping config creation, expecting None values
+    config = get_user_info(args, skip_config_creation=True)
```
Comment (sbillinge) [inline tests/test_tools.py:134]: I am a little confused here....isn't this getting `args` as written?

**Review by sbillinge — COMMENTED** [review]
tbh I find the tests confusing for the ``get_user_info/data`` workflow.  Can we maybe have them written following the new standards that @bobleesj and I worked out?

**Comment by yucongalicechen** [comment]
closed as replaced by #253

**Pull request closed without merging**